### ISSUE TITLE
DB error 1064: syntax error when manually fetching a post from Mastodon

### ISSUE THREAD

**Issue by AlfredSK** [friendica/friendica#9246]
### Expected behavior
When I try to fetch a post from Mastodon via search field it should be imported.

### Actual behavior
When I try to fetch that post via search field `https://mastodon.social/@evolbug/104891984441147017` I get a 'no results' error.

I'm wondering about the "unexpected '@'". I tried other posts from mastodon.social and they were imported as expected.

```
2020-09-20T08:03:25Z index [ERROR]: DB Error {"code":1064,"error":"syntax error, unexpected '@', expecting $end","callstack":"DBA::select, ItemContent::getURIIdListBySearch, Index::content, call_user_func, Page::initContent, Page::run, App::runFrontend","params":"SELECT `uri-id` FROM `item` WHERE (`uri-id` IN (SELECT `uri-id` FROM `item-content` WHERE MATCH (`title`, `content-warning`, `body`) AGAINST ('https:\/\/mastodon.social\/@evolbug\/104891984441147017' IN BOOLEAN MODE)) AND (NOT `private` OR (`private` AND `uid` = 1))) GROUP BY `uri-id` ORDER BY `uri-id` DESC LIMIT 0, 30"} - {"file":"Database.php","line":668,"function":"p","uid":"239f33","process_id":3496247}
2020-09-20T08:03:25Z index [ERROR]: DB Error {"code":1064,"error":"syntax error, unexpected '@', expecting $end","callstack":"DBA::count, ItemContent::countBySearch, Index::content, call_user_func, Page::initContent, Page::run, App::runFrontend","params":"SELECT COUNT(*) AS `count` FROM `item` WHERE (`uri-id` IN (SELECT `uri-id` FROM `item-content` WHERE MATCH (`title`, `content-warning`, `body`) AGAINST ('https:\/\/mastodon.social\/@evolbug\/104891984441147017' IN BOOLEAN MODE)) AND (NOT `private` OR (`private` AND `uid` = 1)))"} - {"file":"Database.php","line":668,"function":"p","uid":"239f33","process_id":3496247}
```

**Related?**
`PHP Notice:  Trying to access array offset on value of type bool in .../src/Database/Database.php on line 1592`

The same thing happens on stable. But w/o the DB error. Import fails.

I tried to disable the full text search and repeated the import attempt. Failed again.

### Originally reported here
https://libranet.de/display/0b6b25a8-105f-6708-0f13-a27545755399

### Steps to reproduce the problem
Manually import `https://mastodon.social/@evolbug/104891984441147017` via search field.
Look at the error info bubble.
Look at the application log for errors.
Look at the PHP log for errors/notices.

### Friendica version you encountered the problem
2020.09-rc
But also fails on stable.

### Friendica source (git, zip)
git

### PHP version
7.4

### SQL version
10.3 MariaDB

**Comment by AlfredSK** [comment]
Uh! There seems to be another issue with importing items now. When successful a query can run for a very long time. That happend when I tried to import the top-level parent post of the above mentioned thread...

**Query_time: 198 sec.**

```
# Time: 200920  8:31:08
# User@Host: xxxxxxxx[xxxxxxxx] @ localhost []
# Thread_id: 788557  Schema: friendica  QC_hit: No
# Query_time: 198.177042  Lock_time: 0.000011  Rows_sent: 25  Rows_examined: 20887471
# Rows_affected: 0  Bytes_sent: 328
# Tmp_tables: 2  Tmp_disk_tables: 0  Tmp_table_sizes: 306825656
# Full_scan: Yes  Full_join: No  Tmp_table: Yes  Tmp_table_on_disk: No
# Filesort: Yes  Filesort_on_disk: No  Merge_passes: 0  Priority_queue: Yes
use friendica;
SET timestamp=1600590668;
SELECT `uri-id` FROM `item` WHERE (`uri-id` IN (SELECT `uri-id` FROM `item-content` WHERE MATCH (`title`, `content-warning`, `body`) AGAINST ('https://pleroma.soykaf.com/notice/9zJov9SQGcisi3pzpQ' IN BOOLEAN MODE)) AND (NOT `private` OR (`private` AND `uid` = 2))) GROUP BY `uri-id` ORDER BY `uri-id` DESC LIMIT 0, 25;
```

Ääähm... and I get search results instead of an import of the post.
"Results for: https://pleroma.soykaf.com/notice/9zJov9SQGcisi3pzpQ"
....
a lot of search result items
....
....

**It looks like the new full text search and the import of remote items do not get along well with each other.**

**Comment by annando** [comment]
The full text search only comes to effect after the search for a post or contact (and only if this wasn't successful).

So when you want to import a post and it cannot be imported since one of the parent posts couldn't be fetched, then the system will perform the search.

**Comment by AlfredSK** [comment]
Yep. That seems to be the case. The failed import could be due to a blocked parent. The original item from the issue has a parent where the instance is on my block list. But the DB error still looks concerning.

The second case - where I tried to import the top level post (from soykaf) - is weird. See the comment. The item seems to be public. It shouldn't fail.(1) And the FT search for the URL is really unexpected (and painfully slow).

Edit: I disabled the FT search for now. That doesn't seem to be production ready. ;-)

(1) `https://pleroma.soykaf.com/notice/9zJov9SQGcisi3pzpQ` is a direct reply to an item where the server is locally blocked. Ok. That explains why it fails. Geez. It's complicated. :-)
